src/main/java/io/r2dbc/postgresql/PostgresqlConnectionConfiguration.java

```diff
@@ -285,6 +360,83 @@ public Builder schema(@Nullable String schema) {
             return this;
         }
 
+        /**
+         * Configure ssl usage.
+         *
+         * @param ssl
+         * @return this {@link Builder}
+         */
+        public Builder ssl(boolean ssl) {
+            this.ssl = ssl;
+            return this;
+        }
+
+        /**
+         * Configure ssl cert.
+         *
+         * @param sslCert
+         * @return this {@link Builder}
+         */
+        public Builder sslCert(@Nullable String sslCert) {
+            this.sslCert = sslCert;
+            return this;
+        }
+
+        /**
+         * Configure ssl HostnameVerifier.
+         *
+         * @param sslHostnameVerifier
+         * @return this {@link Builder}
+         */
+        public Builder sslHostnameVerifier(@Nullable String sslHostnameVerifier) {
+            this.sslHostnameVerifier = sslHostnameVerifier;
+            return this;
+        }
+
+        /**
+         * Configure ssl key.
+         *
+         * @param sslKey
+         * @return this {@link Builder}
+         */
+        public Builder sslKey(@Nullable String sslKey) {
+            this.sslKey = sslKey;
+            return this;
+        }
+
+        /**
+         * Configure ssl mode.
+         *
+         * @param sslMode
+         * @return this {@link Builder}
+         */
+        public Builder sslMode(SSLMode sslMode) {
+            this.sslMode = sslMode;
+            return this;
+        }
+
+        /**
+         * Configure ssl password.
+         *
+         * @param sslPassword
+         * @return this {@link Builder}
+         */
+        public Builder sslPassword(@Nullable String sslPassword) {
```

Comment:
Nit: Let's switch to `Charsequence` to allow for implementations that do not enforce `String` caching in the JVM.

Reply:
SslContextBuilder.keyManager(File, File, String) requires String anyway, so we'll have to convert it to string anyway. Or should I make it CharSequence only in Builder fields?

Reply:
The idea is to help improve the status quo and postpone conversion to String as late as possible. Let’s keep passwords As CharSequence for as long as possible. I’ll take a look at the login password handling (not related to this PR) in the context of a new ticket.